javascript/code-challenges/challenges-10.test.js

```diff
@@ -0,0 +1,384 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named returnTen, takes in a string and uses split and splice to return the last 10 characters from that string as elements of an array.
+
+------------------------------------------------------------------------------------------------ */
+
+function returnTen(str) {
+  return str.split('').splice(-10);
+}
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named findMax that takes in a matrix of positive numbers and returns the number with the highest value.
+
+For example:
+[
+  [1, 3, 4, 5],
+  [4, 5, 6],
+  [23, 5, 5]
+]
+
+return: 23
+------------------------------------------------------------------------------------------------ */
+
+const findMax = (matrix) => {
+  let maxNum = 0;
+  for (let i = 0; i < matrix.length; i++) {
+    for (let j = 0; j < matrix[i].length; j++) {
+      if (maxNum < matrix[i][j]) {
+        maxNum = matrix[i][j];
+      }
```

Comment:
This works, but you'll also want to get in the habit of finding solutions that don't use nested loops. This is generally called "brute force" and while pretty straightforward is often considered the least desirable solution.